### PR TITLE
Add specs for all models

### PR DESCRIPTION
**Pull request by Abe1able**
### In this milestone of the project, I added validations to the models and created specs for them

These are the details:
Added the following validations:

**For the User model:** 
> Name must not be blank.
> PostsCounter must be an integer greater than or equal to zero.

**For the Post model:**
> Title must not be blank.
> Title must not exceed 250 characters.
> CommentsCounter must be an integer greater than or equal to zero.
> LikesCounter must be an integer greater than or equal to zero.

**I added unit specs for all the models' methods and validations.**

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @Abe1able 🙋‍♀️,  

While you made a great effort in this project, I can't review your code.

## Invalid Code Review Request
 
You have submitted a project that has a README that is not professional. Here are key points to consider when documenting your README.

- All main headers from the [README template](https://github.com/microverseinc/readme-template) should be present. ✖️ 

Kindly add and document the `Acknowledgments` section.
image


- There should not be any blank sections that do not have content. ✖️ 

Kindly document the `Show your support` section.
image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+RSpec.describe Post, type: :model do
+  before :each do
+    @user = User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher', posts_counter: 1)
+    @post = Post.new(title: 'Hi', text: 'My world', comments_counter: 0, likes_counter_integer: 0, user_id: 1)
+  end
+
+  it 'checks title presence' do
+    @post.title = nil
+    expect(@post).to_not be_valid
+  end
```
Comment (franclobo) [inline spec/models/post_spec.rb:12]: - [ ] Kindly add this test to validate the maximum number of characters 👇🏽 

> Title must not exceed 250 characters.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Abe1able, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please 🙏🏽 ensure that all the tests passed 👇🏽 

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Favourezeugwa — APPROVED** [review]
## STATUS: APPROVED 🥇 🏆 🍹 

Hi @Abe1able ,

I am Favour 👩🏽 , your reviewer for this PR.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Yayyy! 💃🏽, Congratulations! 🎉 🎉🎉


## Highlights 💯 
- No Linter errors ✔️
- `User` and `Post` model has validations ✔️
- Model specs has **Validation** tests ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
